### PR TITLE
Move `create_dashboard` mehod to DashboardMetadata

### LINKED ISSUE #215: [TECH DEBT] Move `create_dashboards` to `DashboardMetadata.as_lakeview`
It looks like the entire create_dashboard method from here should be converted to a method on `DashboardMetadata`:

```
def as_lakeview(self) -> Dashboard:
  self.validate()
  datasets = self.get_datasets() # make get_datasets private
  layouts = self.get_layouts() # make get_layouts private
  page = Page(
    name=dashboard_metadata.display_name,
    display_name=dashboard_metadata.display_name,
    layout=layouts,
  )
  return Dashboard(datasets=datasets, pages=[page])
```

_Originally posted by @nfx in https://github.com/databrickslabs/lsql/pull/210#discussion_r1673681141_

### PR DESCRIPTION
**Pull request by JCZuurmond**
Resolves #215

### CODE REVIEW

File: src/databricks/labs/lsql/dashboards.py [hunk 1]
```diff
@@ -706,21 +709,33 @@ def replace_database(self, *args, **kwargs) -> "DashboardMetadata":
...
+    def _get_layouts(self) -> list[Layout]:
         """Get the layouts for the dashboard."""
         layouts: list[Layout] = []
         for tile in self.tiles:
             layouts.extend(tile.get_layouts())
         return layouts
 
+    def as_lakeview(self) -> Dashboard:
+        """Create a lakeview dashboard from the dashboard metadata."""
+        datasets = self._get_datasets()
```
Comment (JCZuurmond) [inline src/databricks/labs/lsql/dashboards.py:729]: @nfx : My intuition is to remove the `validate` now as it is a side effect of `as_lakeview` and should be called by the user before `as_lakeview`. We also do not `validate` in the `as_dict`

File: src/databricks/labs/lsql/dashboards.py [hunk 2]
```diff
@@ -422,6 +422,9 @@ def replace_database(
                 The catalog to replace, if None, all catalogs are replaced
             database_to_replace : str | None (default: None)
                 The database to replace, if None, all databases are replaced
+
+        Source :
```
Comment (JCZuurmond) [inline src/databricks/labs/lsql/dashboards.py:426]: Was not in the right place anymore

File: tests/unit/test_dashboards.py [hunk 3]
```diff
@@ -509,90 +509,55 @@ def test_tile_places_tile_below():
     assert tile.position.y == 8
 
 
-def test_dashboards_saves_sql_files_to_folder(tmp_path):
```
Comment (JCZuurmond) [inline tests/unit/test_dashboards.py:512]: Moved the `save_to_folder` tests down to keep the test order somewhat consistent

File: tests/unit/test_dashboards.py [hunk 4]
```diff
@@ -902,159 +849,105 @@ def test_dashboards_overrides_show_empty_title_in_dashboard_yml(tmp_path):
...
+    dashboard = dashboard_metadata.as_lakeview()
 
-    layout = lakeview_dashboard.pages[0].layout
+    layout = dashboard.pages[0].layout
     assert len(layout) == 2
     assert layout[0].position.y < layout[1].position.y
-    ws.assert_not_called()
-
-
-def test_dashboards_creates_dashboard_with_id_collisions_raises_value_error(tmp_path):
```
Comment (JCZuurmond) [inline tests/unit/test_dashboards.py:970]: Removed as validate is removed from `as_lakeview`

**Review by nfx — APPROVED** [review]
lgtm

**Pull request merged**